service/src/main/kotlin/io/provenance/digitalcurrency/consortium/service/CoinMintService.kt

```diff
@@ -16,29 +18,40 @@ class CoinMintService(
 ) {
     private val log = logger()
 
-    fun createEvent(coinMintRecord: CoinMintRecord) {
-        // There should not be any tx events at this point or all event should have an error status
-        val existingEvents: List<TxStatusRecord> =
-            TxStatusRecord.findByTxRequestUuid(coinMintRecord.id.value)
-        check(
-            existingEvents.isEmpty() ||
-                existingEvents.filter { it.status == TxStatus.ERROR }.size == existingEvents.size
-        ) { "Mint/swap contract already called" }
-
-        try {
-            val txResponse = pbcService.mintAndSwap(
-                amount = coinMintRecord.coinAmount.toBigInteger(),
-                address = coinMintRecord.addressRegistration.address
-            ).txResponse
-            log.info("Mint/swap tx hash: ${txResponse.txhash}")
-            TxStatusRecord.insert(
-                txResponse = txResponse,
-                txRequestUuid = coinMintRecord.id.value,
-                type = TxType.MINT_CONTRACT
-            )
-            CoinMintRecord.updateStatus(coinMintRecord.id.value, CoinMintStatus.PENDING_MINT)
-        } catch (e: Exception) {
-            log.error("Mint/swap contract failed; it will retry.", e)
+    fun createEvent(coinMintRecords: List<CoinMintRecord>) {
+        coinMintRecords.mapNotNull { coinMintRecord ->
+            // There should not be any tx events at this point or all event should have an error status
+            val existingEvents: List<TxStatusRecord> =
+                TxStatusRecord.findByTxRequestUuid(coinMintRecord.id.value)
+            if (!(existingEvents.isEmpty() || existingEvents.filter { it.status == TxStatus.ERROR }.size == existingEvents.size)) {
+                log.error("Mint contract already called")
```

Comment:
is it just me or should something like this never happen? wondering why we even handle it beyond exceptioning and finding the bug that caused it?

Reply:
we actually did have this happen in the old omnibus this month on a burn and you are right - it should have just thrown an exception and not updated the status on the omnibus records because they had already been completed successfully.